### PR TITLE
github workflows: repo-wide auto-format

### PR DESCRIPTION
**Pull request by postamar**
This PR follows up on https://github.com/airbytehq/airbyte/pull/29786 and changes the existing `gradle.yml` github workflow to insert a new `format` job near the start of the job dependency graph. This job auto-commits the results of `./gradlew format` applied at the root of the repo. This is in continuation of the existing workflow behaviour where `format` was applied (and the results auto-committed) for select modules only.

In rhe process of building this workflow, I found that the existing `./gradlew format` was far too slow to be practical. This is because the python formatters rely on each project's venv. This isn't necessary when it comes to just formatting, so I modified the gradle `format` task to use a "global" isort and black call using a new venv at the root of the repo. The resulting `format` job only takes about 5 minutes now.

### CODE REVIEW

**Comment by postamar** [comment]
Heh, looks like this isn't quite ready to merge yet, `airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py` is getting clobbered by two different formatting rules. I'll investigate.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py
Comment (aaronsteers) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py:19]: This file is preferring single quotes, while others later in the PR are preferring double quotes. Might be worth investigating what is driving the different behavior.

I think general recommended default with Blank is to prefer double quotes except when single quotes are cleaner. 

https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings

Comparison file later in the PR:

`resources/examples/airflow/dags/dag_airbyte_example.py`

Comment (aaronsteers) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py]: This is resolved in the subsequent commit: 

https://github.com/airbytehq/airbyte/pull/29798/commits/db27f271b2b7a8114e125bca8dcd50d3dea952aa

**Review by aaronsteers — APPROVED** [review]
My approval shouldn't be viewed as sufficient on its own, but I'm approving nonetheless for my part. I'm very excited to see this merged, as I've several times been distracted by these formatting issues when onboarding to the python code.

Long-term considerations:

**Regarding line length**, I'd go further and suggest to reduce mex line length from `140` to `120` in the root `pyproject.toml`, which matches what we were suggesting on the Java side. If there were appetite to go further, for Python my personal preference is `88` or `100` characters max, as those widths are better for side-by-side diffs and reduce left-to-right eye movement when reading large amounts of code.

That said, changes to line length can be moved to a follow-on discussion, but I've put my thoughts here for completeness.

**Regarding CI auto-fix**, I think it is okay to keep auto-format in place and to expand to the full repo - especially once this one-time update is applied. This should keep us in a good place formatting wise and prevent future drift. My preference long-term would be to move this to a GitHub PR slash command action (`/autoformat` or similar) - as that brings similar convenience in an opt-in/on-demand capacity. Combined with raising a failure in CI, we're in a good place to keep code style consistent going forward 👍 .

As with the line length topic, I wouldn't want to block this PR for any of those long-term DX considerations.

**Comment by postamar** [comment]
Thanks for taking a look @aaronsteers! I'm deliberately focusing on adding the format automation first before changing the format styles themselves.

Once this merges someone with admin permissions should add a required check on the repo to the effect that the formatting job succeeds. All it takes to fail is a duplicate key in a yaml file, which seems to be happening with some frequency lately.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py
Comment (postamar) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/models/declarative_component_schema.py]: As it turns out the issue was a bit more complicated and involved code generation. It's since been fixed. Furthermore, I now have the python license task run globally as well.

**Pull request merged**
